R/check-default-branch.R

```diff
@@ -0,0 +1,34 @@
+#' Ensure that default GitHub branch is not "master"
+#'
+#' The `$info$git$branch` value is taken by default from GitHub as long as
+#' "DESC" has a remote URL. It is only taken from local git if not remote
+#' GitHub URL can be identified.`
+#'
+#' @param checks A 'pkgcheck' object with full \pkg{pkgstats} summary and
+#' \pkg{goodpractice} results.
+#' @return TRUE if default GitHub branch is "master"; otherwise FALSE
+#' @noRd
+pkgchk_branch_is_master <- function (checks) {
+
+    ret <- FALSE
+    if (length (checks$info$git) > 0L) {
+        ret <- checks$info$git$branch == "master"
```

Comment:
if checks$info$git is of 0 length is it NULL? If so could the whole function body use rlang::`%||%`

```r
(checks$info$git$branch %||% "main") == "master"

```

or in two lines


```r
default_branch <- checks$info$git$branch %||% "main"

default_branch == "master"

```

Reply:
No, it is not `NULL`, so that's can't be used, alas. It is *generally* an empty list, but I'm also not sure whether it might not also be possible to generate `NULL`? Either way, I think being explicit it hefpful there.